### ISSUE TITLE
Add Transition capabilities 

### ISSUE THREAD

**Issue by aproposnix** [trynoice/android-app#144]
### Add sound transitions
As a user, I would like to imitate rain variability. This means I would like to start off with a light rain sound and then eventually transition to a heavy rain sound and vice versa. 

### Use Case
I propose to add transition effects into the app. 

A user can select multiple sounds and group them as 'transition sounds'. This means the app will cycle through those sounds (from the group created) after X amount of time for a single track (each group = 1 track)

So in my example, I would create a group called "Rain Sounds". in that group I would select 3 different sounds: light rain, medium rain and heavy rain. After creating the group, I could then go ahead and add other tracks separate from the group (as you normally can now). 

The app will randomly start one of the 3 rain tracks which were added to the group above. For example, it starts with light rain. After an hour or so, Noice will transition to one of the other two rain tracks. It will do this until either the Noice timer goes of or the user turns Noice off.

**Comment by ashutoshgngwr** [comment]
@aproposnix This is a nice suggestion and I've been thinking about something similar along the lines.

I was thinking of taking away user's control over the variety of sound. What I mean by that is instead of showing the user three version of rain sound, we show them just one version. This version in background takes cares of transitioning between the three versions randomly. In addition, we can also give these sounds a randomness slider (like we do have right now for non-looping sounds). This slider can be used to tune how often does the user want it to transition.

As for your version of the idea, the only complication I see on my end is that it will bring a certain complexity to the user experience in the app. 

Please let me know you thoughts on this.

**Comment by aproposnix** [comment]
I do agree that it could bring more complexity to the app, but only if you don't design it right. One way to avoid the complexity for average users would be simply to add "advanced controls". From here you would be able to create the groups. Once created, the groups appear as sound banks to pick from. 

I think the important thing is to provide clear documentation to the user. Maybe a feature tour?

**Comment by ashutoshgngwr** [comment]
I have an idea. What if you could horizontally swipe each sound and group them using colors.

On swiping on a sound, you have following options
1. Add sound to a group if it is not part of one already
   - On swiping, you will be displayed colored circle representing existing groups in current playback
   - Along with colored circles, you'll have an option to create a new group
2. If sound is already added to the group, you will have options to change its group

Additionally, each sound can have a colored stripe in its normal view that shows is group. This will need to be documented ofcourse.

**Comment by aproposnix** [comment]
To be honest, it sounds kind of complicated and a potential risk for visual clutter.  Maybe I don't understand it fully though. 

Why not allow the use to long press a sound which will open a dialogue, select group sounds... the user selects the sounds, names it and then saves it. Once taken back to the sound list, the new group will be visible as a new sound bank.

**Comment by ashutoshgngwr** [comment]
Okay. I was missing the part where group are also shown as sounds. Let me think a little bit more about it. I'll definitely share my final thoughts on this before heading into implementation.

**Comment by ashutoshgngwr** [comment]
> To be honest, it sounds kind of complicated and a potential risk for visual clutter. Maybe I don't understand it fully though.
> 
> Why not allow the use to long press a sound which will open a dialogue, select group sounds... the user selects the sounds, names it and then saves it. Once taken back to the sound list, the new group will be visible as a new sound bank.



> I had an idea about the UX since our last exchange. What if you could horizontally swipe each sound and group them using colors. This actually builds on what you have suggested.
> 
> On swiping a sound horizontally, you can add sound to a group if it is not part of one already. After swiping, you will be displayed colored circle representing existing groups in current playback. Along with colored circles, you'll have an option to create a new group. If sound is already part of a group, the corresponding colored circle will have a control to remove the sound from it.
> 
> Additionally, each sound can have a colored stripe in its normal view that shows is group. This will need to be documented ofcourse.

Going with these two combined. Not sure about the end result. Let's hope this works out.

**Comment by ashutoshgngwr** [comment]
I had started to work on this a couple of times over the past few months and for one reason or another, I haven't been able to complete the implementation. I am marking this as "up for grabs" so that if anyone is available to take this over, they should be able to. I am also deleting the working branch that I had since I have mostly forgotten about what I had finished and what was still remaining.
